tests/unit/mesh/test_ncmesh.cpp

```diff
@@ -422,103 +409,243 @@ TEST_CASE("FaceEdgeConstraint",  "[Parallel], [NCMesh]")
-      stmp.GeneralRefinement(serial_refines);
-      REQUIRE(stmp.GetNE() == 1 + 8 - 1 + 8); // 16 elements
+      REQUIRE(pmesh.GetGlobalNE() == 8 + 1);
+      REQUIRE(smesh.GetNE() == 8 + 1);
 
-      if (Mpi::WorldRank() == refining_rank)
+      // Each pair of indices here represents sequential element indices to refine.
+      // First the i element is refined, then in the resulting mesh the j element is
+      // refined. These pairs were arrived at by looping over all possible i,j pairs and
+      // checking for the addition of a face-edge constraint.
+      std::vector<std::pair<int,int>> indices{{2,13}, {3,13}, {6,2}, {6,3}};
+
+      // Rank 0 has all but one element in the parallel mesh. The remaining element
+      // is owned by another processor if the number of ranks is greater than one.
+      for (const auto &ij : indices)
       {
-         refines[0] = j;
+         int i = ij.first;
+         int j = ij.second;
+         if (Mpi::WorldRank() == refining_rank)
+         {
+            refines[0] = i;
+         }
+         // Inform all ranks of the serial mesh
+         serial_refines[0] = pmesh.GetGlobalElementNum(i);
+         MPI_Bcast(&serial_refines[0], 1, MPI_INT, 0, MPI_COMM_WORLD);
+
+         ParMesh tmp(pmesh);
+         tmp.GeneralRefinement(refines);
+
+         REQUIRE(tmp.GetGlobalNE() == 1 + 8 - 1 + 8); // 16 elements
+
+         Mesh stmp(smesh);
+         stmp.GeneralRefinement(serial_refines);
+         REQUIRE(stmp.GetNE() == 1 + 8 - 1 + 8); // 16 elements
+
+         if (Mpi::WorldRank() == refining_rank)
+         {
+            refines[0] = j;
+         }
+         // Inform all ranks of the serial mesh
+         serial_refines[0] = tmp.GetGlobalElementNum(j);
+         MPI_Bcast(&serial_refines[0], 1, MPI_INT, 0, MPI_COMM_WORLD);
+
+         ParMesh ttmp(tmp);
+         ttmp.GeneralRefinement(refines);
+
+         REQUIRE(ttmp.GetGlobalNE() == 1 + 8 - 1 + 8 - 1 + 8); // 23 elements
+
+         Mesh sttmp(stmp);
+         sttmp.GeneralRefinement(serial_refines);
+         REQUIRE(sttmp.GetNE() == 1 + 8 - 1 + 8 - 1 + 8); // 23 elements
+
+         // Loop over interior faces, fill and check face transform on the serial.
+         for (int iface = 0; iface < sttmp.GetNumFaces(); ++iface)
+         {
+            const auto face_transform = sttmp.GetFaceElementTransformations(iface);
+            CHECK(face_transform->CheckConsistency(0) < 1e-12);
+         }
+
+         for (int iface = 0; iface < ttmp.GetNumFacesWithGhost(); ++iface)
+         {
+            const auto face_transform = ttmp.GetFaceElementTransformations(iface);
+            CHECK(face_transform->CheckConsistency(0) < 1e-12);
+         }
+
+         // Use P4 to ensure there's a few fully interior DOF.
+         {
+            auto error = CheckL2Projection(ttmp, sttmp, 4, exact_soln);
+            double constexpr tol = 1e-9;
+            // std::cout << "R" << Mpi::WorldRank() << ": " << error[0] << " " << error[1] << std::endl;
```

Comment:
There are 2 new `cout` lines that can be removed.

Reply:
Fixed in https://github.com/mfem/mfem/pull/3926/commits/c27bec417c62cb6dd7167dfa3939bbc1142fe31a